File: manual/data/java/tests/test_cvxpy_norm.java
package com.example;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class test_cvxpy_norm {

    @Test
    void test_norm_type() {
        String vector = "[1, 2, 3]";
        String normValue = new function_cvxpy_norm().compute_norm(vector, 2);
        assertNotNull(normValue);
        assertTrue(normValue instanceof String);
    }

    @Test
    void test_norm_computation_2_norm() {
        String vector = "[3, 4, 0]";
        String normValue = new function_cvxpy_norm().compute_norm(vector, 2);
        assertEquals("5.0", normValue);
    }

    @Test
    void test_norm_computation_1_norm() {
        String vector = "[1, -2, 3]";
        String normValue = new function_cvxpy_norm().compute_norm(vector, 1);
        assertEquals("6.0", normValue);
    }

    @Test
    void test_norm_computation_inf_norm() {
        String vector = "[-1, 2, -3, 4]";
        String normValue = new function_cvxpy_norm().compute_norm(vector, Integer.MAX_VALUE);
        assertEquals("4.0", normValue);
    }

    @Test
    void test_norm_zero_vector() {
        String vector = "[0, 0, 0, 0, 0]";
        String normValue = new function_cvxpy_norm().compute_norm(vector, 2);
        assertEquals("0.0", normValue);
    }
}

File: manual/code/test_suites/python/python/test_cvxpy_norm.py
import unittest
from function_cvxpy_norm import CVXPYNormFunction

class TestCVXPYNormFunction(unittest.TestCase):

    def test_norm_type(self):
        vector = "[1, 2, 3]"  
        norm_value = CVXPYNormFunction().compute_norm(vector)
        self.assertIsInstance(norm_value, str)  

    def test_norm_computation_2_norm(self):
        vector = "[3, 4, 0]"
        norm_value = CVXPYNormFunction().compute_norm(vector, p=2)
        self.assertEqual(norm_value, "5.0") 

    def test_norm_computation_1_norm(self):
        vector = "[1, -2, 3]"  
        norm_value = CVXPYNormFunction().compute_norm(vector, p=1)
        self.assertEqual(norm_value, "6.0") 

    def test_norm_computation_inf_norm(self):
        vector = "[-1, 2, -3, 4]" 
        norm_value = CVXPYNormFunction().compute_norm(vector, p='inf')
        self.assertEqual(norm_value, "4.0")  

    def test_norm_zero_vector(self):
        vector = "[0, 0, 0, 0, 0]"  
        norm_value = CVXPYNormFunction().compute_norm(vector, p=2)
        self.assertEqual(norm_value, "0.0") 

if __name__ == "__main__":
    unittest.main()
